compiler/rustc_hir/src/definitions.rs

```diff
@@ -155,6 +155,32 @@ pub struct DisambiguatedDefPathData {
     pub disambiguator: u32,
 }
 
+impl DisambiguatedDefPathData {
+    pub fn format<W: Write>(&self, writer: &mut W, verbose: bool) -> fmt::Result {
```

Comment:
There's a more "user-friendly" way to do this, that would have this signature:
```rust
fn display(&self, verbose: bool) -> impl fmt::Display
```

If that seems a bit much, I'd call this `fmt_maybe_verbose` or something and keep the current signature (though you can have `write: &mut impl Write` instead of the explicit generic).